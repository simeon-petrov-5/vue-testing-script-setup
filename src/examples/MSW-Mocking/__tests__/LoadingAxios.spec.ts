import { flushPromises, mount } from '@vue/test-utils';
import LoadingAxios from '../LoadingAxios.vue';

import { beforeAll, afterAll, afterEach } from 'vitest';
import { setupServer } from 'msw/node';
import { rest } from 'msw';
import mockedCharacters from '../../../mock-data/characters.json';

export const restHandlers = [
  rest.get('https://rickandmortyapi.com/api/character', (req, res, ctx) => {
    return res(ctx.status(200), ctx.json(mockedCharacters));
  })
];

const server = setupServer(...restHandlers);

const createWrapper = () => {
  const wrapper = mount(LoadingAxios);
  return wrapper;
};

describe('LoadingAxios.vue', () => {
  beforeAll(() => server.listen({ onUnhandledRequest: 'error' }));
  afterAll(() => server.close());
  afterEach(() => server.resetHandlers());

  it('should match snapshot', async () => {
    const wrapper = createWrapper();
    expect(wrapper.text()).toContain('Dummy Loading page');

    // await nextTick();
    await flushPromises();
    await flushPromises();
    await flushPromises();

    // await nextTick();
    // await nextTick();

    expect(wrapper.element).toMatchSnapshot();
    expect(wrapper.text()).not.toContain('Page is loading');
  });
});
