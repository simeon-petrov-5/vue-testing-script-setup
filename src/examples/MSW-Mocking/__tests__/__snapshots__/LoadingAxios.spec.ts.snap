// Vitest Snapshot v1

exports[`LoadingAxios.vue > should match snapshot 1`] = `
<div
  data-v-app=""
>
  
  <h1>
    Dummy Loading page
  </h1>
  <!--v-if-->
  <ul>
    
    <li>
      Rick Sanchez
    </li>
    <li>
      Morty Smith
    </li>
    <li>
      Summer Smith
    </li>
    <li>
      Beth Smith
    </li>
    <li>
      Jerry Smith
    </li>
    <li>
      Abadango Cluster Princess
    </li>
    <li>
      Abradolf Lincler
    </li>
    <li>
      Adjudicator Rick
    </li>
    <li>
      Agency Director
    </li>
    <li>
      Alan Rails
    </li>
    <li>
      Albert Einstein
    </li>
    <li>
      Alexander
    </li>
    <li>
      Alien Googah
    </li>
    <li>
      Alien Morty
    </li>
    <li>
      Alien Rick
    </li>
    <li>
      Amish Cyborg
    </li>
    <li>
      Annie
    </li>
    <li>
      Antenna Morty
    </li>
    <li>
      Antenna Rick
    </li>
    <li>
      Ants in my Eyes Johnson
    </li>
    
  </ul>
  
</div>
`;
